addons/mail/controllers/thread.py

```diff
@@ -189,8 +189,12 @@ def mail_message_post(self, thread_model, thread_id, post_data, context=None, **
                 'last_used': datetime.now(),
                 'ids': canned_response_ids,
             })
-        # TDE todo: should rely on '_get_mail_message_access'
-        thread = self._get_thread_with_access(thread_model, thread_id, mode=request.env[thread_model]._mail_post_access, **kwargs)
+        # creating a message require an access on model, defined by '_get_mail_message_access'
+        # default behavior being to rely on _mail_post_access
+        access_mode = request.env[thread_model].sudo().browse(thread_id)._get_mail_message_access('create')
```

Comment:
I don't think we want to allow non-thread here. I believe previous code would crash here (as the method didn't exist on other models?). Is that still preventing it?

Reply:
I guess it will still crash on `_get_thread_with_access`.